src/components/website-management/website-form-components/customization/customization-components/BannerTemplateTable.js

```diff
@@ -0,0 +1,121 @@
+import React, { useContext, useEffect, useState } from "react";
+import { Switch } from "@mui/material";
+import { styled } from "@mui/material/styles";
+import TableCell, { tableCellClasses } from "@mui/material/TableCell";
+import TableRow from "@mui/material/TableRow";
+import Typography from "@mui/material/Typography";
+import TableContainer from "@mui/material/TableContainer";
+import Paper from "@mui/material/Paper";
+import Table from "@mui/material/Table";
+import TableHead from "@mui/material/TableHead";
+import TableBody from "@mui/material/TableBody";
+import axios from "axios";
+import { BannerContext } from "./BannerContext";
+
+const BannerTemplateTable = () => {
+  const [templates, settemplates] = useState([]);
+  const [selectedTemplateId, setSelectedTemplateId] = useState("");
+  const passedId = useContext(BannerContext).websiteId;
+
+  const loadTemplates = async () => {
+    const websiteID = passedId;
+    const result = await axios.get(
+      `http://localhost:8080/api/template/getAllTemplates/${websiteID}`
+    );
+    settemplates(result.data);
+  };
+
+  useEffect(() => {
+    loadTemplates();
+  }, []);
+
+  const handleSwitchChange = (selectedTemplate) => {
+    const updatedTemplates = templates.map((template) => {
+      if (template.templateId === selectedTemplate.templateId) {
+        // Set the selected switch to true
+        axios.put(
+          `http://localhost:8080/api/template/templateDefault/${selectedTemplate.templateId}`,
+          { templateDefault: "true" }
+        );
+        setSelectedTemplateId(selectedTemplate.templateId);
+        return { ...template, templateDefault: true }; // Update the template with templateDefault: true
```

Comment:
Let's put the comment above the line of code for better readability.

```suggestion
        // Update the template with templateDefault: true
        return { ...template, templateDefault: true }; 
```

Reply:
fixed